code/modules/tgui_input/say_modal/tgui_say_modal.dm

```diff
@@ -0,0 +1,137 @@
+/** Assigned say modal of the client */
+/client/var/datum/tgui_say/tgui_say
+
+/**
+ * Creates a JSON encoded message to open TGUI say modals properly.
+ *
+ * Arguments:
+ * channel - The channel to open the modal in.
+ * Returns:
+ * string - A JSON encoded message to open the modal.
+ */
+/client/proc/tgui_say_create_open_command(channel)
+	var/message = TGUI_CREATE_MESSAGE("open", list(
+		channel = channel,
+	))
+	return "\".output tgui_say.browser:update [message]\""
+
+/**
+ * The tgui say modal. This initializes an input window which hides until
+ * the user presses one of the speech hotkeys. Once something is entered, it will
+ * delegate the speech to the proper channel.
+ */
+/datum/tgui_say
+	/// The user who opened the window
+	var/client/client
+	/// Injury phrases to blurt out
+	var/list/hurt_phrases = list("GACK!", "GLORF!", "OOF!", "AUGH!", "OW!", "URGH!", "HRNK!")
```

Comment:
Some of these are kind of silly, and I'm not entirely against them, but I think it would be good to have more realistic ones. ("AGH!", "ARGH!!", "ACK!!", "OUF!" are some I've seen on other servers).

(Also, should this list be static?)